acceptance/tests/facts/windows_os.rb

```diff
@@ -0,0 +1,35 @@
+test_name "Test facter os reports correct os.windows.system32" do
+
+  confine :to, :platform => /windows/
+
+  agents.each do |agent|
+
+    step "Install Windows remote desktop feature" do
+      os_type = on(agent, powershell("facter os.windows.installation_type")).stdout.chomp
+      if os_type == "Server"
+        on(agent, powershell("facter os.windows.system32")) do |facter_before_output|
+          assert_equal("C:\\Windows\\system32", facter_before_output.stdout.chomp, 'Before windows feature installation, should be C:\\Windows\\system32')
+        end
+        on(agent, powershell("Add-WindowsFeature –Name RDS-RD-Server –IncludeAllSubFeature"))
+        on(agent, powershell("'Import-module servermanager ; (Get-WindowsFeature -name RDS-RD-Server).Installed'")) do |installed_feature|
+          assert_equal("True", installed_feature.stdout.chomp, 'Result should be true for installed RDS-RD-Server feature')
+        end
+        agent.reboot
+      end
+    end
+
+    step "Verify facter os reports correct system32 variable" do
+      on(agent, powershell("facter os.windows.system32")) do |facter_output|
+        assert_equal("C:\\Windows\\system32", facter_output.stdout.chomp, 'Result should be C:\\Windows\\system32')
+      end
+    end
+
+    teardown do
+      is_server = on(agent, powershell("'Import-module servermanager ; (Get-WindowsFeature -name RDS-RD-Server).Installed'"))
```

Comment:
can't we move `os_type = on(agent, powershell("facter os.windows.installation_type")).stdout.chomp` outside of any `step` and reuse it here and in the first step? i'm thinking on reducing the system calls so the test runs faster.
Would `on(agent, powershell("Remove-WindowsFeature –Name RDS-RD-Server"))` fail if RDS is not enabled?

Reply:
on(agent, powershell("Remove-WindowsFeature –Name RDS-RD-Server")) it fails,
os_type = on(agent, powershell("facter os.windows.installation_type")).stdout.chomp tests for something else, why use it here ?

Reply:
Because you can only enable or disable RDS only on an `os_type == server`. You are testing that an os is server by running 2 different commands `on(agent, powershell("facter os.windows.installation_type"))` and `on(agent, powershell("'Import-module servermanager ; (Get-WindowsFeature -name RDS-RD-Server).Installed'"))`